lightning/src/ln/interactivetxs.rs

```diff
@@ -513,6 +533,175 @@ impl InteractiveTxSigningSession {
+					return Err(APIError::APIMisuseError { err });
+				}
+
+				let sighash = cache
+					.p2wpkh_signature_hash(
+						input_idx,
+						script_pubkey,
+						prev_output.value,
+						EcdsaSighashType::All,
+					)
+					.map_err(|_| {
+						debug_assert!(false, "Funding transaction sighash should be calculable");
+						APIError::APIMisuseError {
+							err: "The transaction sighash could not be calculated".to_string(),
+						}
+					})?;
+				let msg = Message::from_digest_slice(&sighash[..])
+					.expect("Sighash is a SHA256 which is 32 bytes long");
+				secp_ctx.verify_ecdsa(&msg, &sig.signature, &pubkey).map_err(|_| {
+					let err = format!("Failed signature verification for input at index {input_idx} for P2WPKH spend");
+					APIError::APIMisuseError { err }
+				})?;
+
+				continue 'inputs;
+			}
+
+			// P2TR key path spend witness includes signature and optional annex
+			if script_pubkey.is_p2tr() && witness.len() == 1 {
+				let pubkey = match script_pubkey.instructions().nth(1) {
+						Some(Ok(bitcoin::script::Instruction::PushBytes(push_bytes))) => {
+							XOnlyPublicKey::from_slice(push_bytes.as_bytes())
+						},
+						_ => {
+							let err = format!("The scriptPubKey of the previous output for input at index {input_idx} for a P2TR key path spend is invalid");
+							return Err(APIError::APIMisuseError { err })
+						},
+					}.map_err(|_| {
+						let err = format!("The scriptPubKey of the previous output for input at index {input_idx} for a P2TR key path spend has an invalid public key");
+						APIError::APIMisuseError { err }
+					})?;
+
+				let sig = bitcoin::taproot::Signature::from_slice(&witness[0]).map_err(|_| {
+					let err = format!("The witness for input at index {input_idx} for a P2TR key path spend has an invalid signature");
+					APIError::APIMisuseError { err }
+				})?;
+				if !matches!(sig.sighash_type, TapSighashType::Default | TapSighashType::All) {
+					let err = format!("Signature does not use SIGHASH_DEFAULT or SIGHASH_ALL for input at index {input_idx} for P2TR key path spend");
+					return Err(APIError::APIMisuseError { err });
+				}
+
+				let sighash = cache
+					.taproot_key_spend_signature_hash(input_idx, &all_prevouts, sig.sighash_type)
+					.map_err(|_| {
+						debug_assert!(false, "Funding transaction sighash should be calculable");
+						APIError::APIMisuseError {
+							err: "The transaction sighash could not be calculated".to_string(),
+						}
+					})?;
+				let msg = Message::from_digest_slice(&sighash[..])
+					.expect("Sighash is a SHA256 which is 32 bytes long");
+				secp_ctx.verify_schnorr(&sig.signature, &msg, &pubkey).map_err(|_| {
+					let err = format!("Failed signature verification for input at index {input_idx} for P2TR key path spend");
+					APIError::APIMisuseError { err }
+				})?;
+
+				continue 'inputs;
+			}
+
+			// P2WSH - No validation just sighash checks
+			if script_pubkey.is_p2wsh() {
+				'elements: for element in witness {
```

Comment:
Are these labels even necessary if you never `continue` to an outer loop?

Reply:
Not necessary, just thought it may be clearer labelling them.